### PR TITLE
[x86/Linux] Fix marshalling struct with 64-bit types

### PR DESCRIPTION
**Pull request by kbaladurin**
The System V ABI for i386 defines 4-byte alignment for 64-bit types.

### CODE REVIEW

**Comment by kbaladurin** [comment]
StructPacking test has still failed with this patch:
```
FAILED   - Interop/StructPacking/StructPacking/StructPacking.sh
               BEGIN EXECUTION
               /media/kbaladurin/data/dotnet/clr_overlay_x86_test/corerun StructPacking.exe
               Unexpected Offset for AutoLayoutDefaultPacking`1[System.Double].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutDefaultPacking`1[System.Double].Value.
                    Expected: 4; Actual: 0
               
               Unexpected Offset for AutoLayoutMinPacking`1[System.Double].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutMinPacking`1[System.Double].Value.
                    Expected: 4; Actual: 0
               
               Unexpected Offset for AutoLayoutMaxPacking`1[System.Double].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutMaxPacking`1[System.Double].Value.
                    Expected: 4; Actual: 0
               
               Unexpected Offset for AutoLayoutDefaultPacking`1[System.Int64].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutDefaultPacking`1[System.Int64].Value.
                    Expected: 4; Actual: 0
               
               Unexpected Offset for AutoLayoutMinPacking`1[System.Int64].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutMinPacking`1[System.Int64].Value.
                    Expected: 4; Actual: 0
               
               Unexpected Offset for AutoLayoutMaxPacking`1[System.Int64].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutMaxPacking`1[System.Int64].Value.
                    Expected: 4; Actual: 0
               
               Unexpected Offset for AutoLayoutDefaultPacking`1[System.UInt64].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutDefaultPacking`1[System.UInt64].Value.
                    Expected: 4; Actual: 0
               
               Unexpected Offset for AutoLayoutMinPacking`1[System.UInt64].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutMinPacking`1[System.UInt64].Value.
                    Expected: 4; Actual: 0
               
               Unexpected Offset for AutoLayoutMaxPacking`1[System.UInt64].Byte.
                    Expected: 0; Actual: 8
               Unexpected Offset for AutoLayoutMaxPacking`1[System.UInt64].Value.
                    Expected: 4; Actual: 0
               
               Expected: 100
               Actual: 0
               END EXECUTION - FAILED
```
The problem is that runtime places largest fields first for structures with auto layout (`vm/methodtablebuilder.cpp:8007`):
```cpp
        // Place fields, largest first
        for (i = MAX_LOG2_PRIMITIVE_FIELD_SIZE; (signed int) i >= 0; i--)
        {
            if (bmtFP->NumInstanceFieldsOfSize[i] == 0)
                continue;

            // Align instance fields if we aren't already
#ifdef FEATURE_64BIT_ALIGNMENT
            DWORD dwDataAlignment = 1 << i;
#else
            DWORD dwDataAlignment = min(1 << i, DATA_ALIGNMENT);
#endif
            dwCumulativeInstanceFieldPos = (DWORD)ALIGN_UP(dwCumulativeInstanceFieldPos, dwDataAlignment);

            // Fields of this size start at the next available location
            bmtFP->InstanceFieldStart[i] = dwCumulativeInstanceFieldPos;
            dwCumulativeInstanceFieldPos += (bmtFP->NumInstanceFieldsOfSize[i] << i);

            // Reset counters for the loop after this one
            bmtFP->NumInstanceFieldsOfSize[i]  = 0;
        }
```
but for some reason test assumes that in case of x86/Linux it doesn't rearranges fields. 
I think we should fix test in this case. What do you think about it? Thank you!

**Comment by kbaladurin** [comment]
cc @Buyduck 
@jkotas @janvorli PTAL

File: src/vm/fieldmarshaler.h [hunk 1]
```diff
@@ -1626,7 +1626,14 @@ class FieldMarshaler_Copy8 : public FieldMarshaler
     UNUSED_METHOD_IMPL(VOID UpdateNativeImpl(OBJECTREF* pCLRValue, LPVOID pNativeValue, OBJECTREF *ppCleanupWorkListOnStack) const)
     UNUSED_METHOD_IMPL(VOID UpdateCLRImpl(const VOID *pNativeValue, OBJECTREF *ppProtectedCLRValue, OBJECTREF *ppProtectedOldCLRValue) const)
 
+#if defined(_TARGET_X86_) && defined(UNIX_X86_ABI)
+    // The System V ABI for i386 defines 4-byte alignment for 64-bit types.
+
```
Comment (luqunl) [inline src/vm/fieldmarshaler.h]: nit: Could you remove this empty line?

Comment (jkotas) [reply src/vm/fieldmarshaler.h]: (in other places too)

Comment (kbaladurin) [reply src/vm/fieldmarshaler.h]: Done. Thank you!

**Pull request merged**